lib/gtfs.ex

```diff
@@ -73,16 +73,38 @@ defmodule Gtfs do
   end
 
   def handle_call({:timepoints_on_route, route_id}, _from, {:loaded, gtfs_data} = state) do
-    timepoint_ids =
+    route_patterns_by_direction =
       gtfs_data.route_patterns
       |> Enum.filter(fn route_pattern -> route_pattern.route_id == route_id end)
-      |> Enum.map(fn route_pattern -> route_pattern.representative_trip_id end)
-      |> Enum.flat_map(fn trip_id -> gtfs_data.stop_times[trip_id] end)
-      |> Enum.map(fn stop_time -> stop_time.timepoint_id end)
-      |> Enum.uniq()
-      |> List.delete("")
+      |> Enum.group_by(fn route_pattern -> route_pattern.direction_id end)
+
+    timepoints_by_direction =
+      Helpers.map_values(route_patterns_by_direction, fn route_patterns ->
```

Comment:
similarly, `Helpers.map_values(route_patterns_by_direction, &timepoints_for_route_patterns(&1, gtfs_data))`

Reply:
Same as above. I have a slight personal preference for named arguments.